xarray/tests/test_backends.py

```diff
@@ -2171,10 +2172,11 @@ def create_store(self):
         with create_tmp_file() as tmp_file:
             yield backends.H5NetCDFStore(tmp_file, "w")
 
+    # TODO: Reduce num_warns by 1 when h5netcdf is updated to not issue the make_scale warning
     @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
     @pytest.mark.parametrize(
         "invalid_netcdf, warns, num_warns",
-        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
```

Comment:
Maybe we could just remove the `assert len(record) == num_warns` line below? I think pytest already verifies that we get a warning with `pytest.warns`.

We can use the `match` argument if we want to keep it more specific: https://docs.pytest.org/en/latest/warnings.html#warns